archlinuxcn/devtools-archlinuxcn/0004-Handle-ArchLinuxARM-server-URL-in-arch-nspawn.patch

```diff
@@ -0,0 +1,53 @@
+From a752bb326bb91f61fe6e03457f2642f45bd1d94f Mon Sep 17 00:00:00 2001
+From: Yichao Yu <yyc1992@gmail.com>
+Date: Sat, 4 Jun 2022 11:19:13 -0400
+Subject: [PATCH 4/4] Handle ArchLinuxARM server URL in arch-nspawn
+
+---
+ arch-nspawn.in | 20 +++++++++++++++++---
+ 1 file changed, 17 insertions(+), 3 deletions(-)
+
+diff --git a/arch-nspawn.in b/arch-nspawn.in
+index b6c9cb1..2e263d5 100644
+--- a/arch-nspawn.in
++++ b/arch-nspawn.in
+@@ -54,11 +54,25 @@ if (( ${#cache_dirs[@]} == 0 )); then
+ fi
+ 
+ # shellcheck disable=2016
+-host_mirrors=($(pacman-conf --repo extra Server 2> /dev/null | sed -r 's#(.*/)extra/os/.*#\1$repo/os/$arch#'))
++pacman_arch=$(pacman-conf Architecture)
++if [[ $pacman_arch = arm ]] || [[ $pacman_arch = armv6h ]] || [[ $pacman_arch = armv7h ]] || [[ $pacman_arch = aarch64 ]]; then
++	is_arm=1
++else
++	is_arm=0
++fi
++if ((is_arm)); then
```

Comment:
I think it is better not to use intermediate variables for such a conditional judgment that has a clear meaning and is only used once?

Reply:
That's fair. An earlier version had this used elsewhere before I decided that it's easier to just set all the variables here....